honeybee_radiance/cli/translate.py

```diff
@@ -0,0 +1,197 @@
+            'No Model JSON file found at {}.'.format(model_json)
+
+        # set the default folder if it's not specified
+        if folder is None:
+            folder = os.path.split(os.path.abspath(model_json))[0]
+
+        # re-serialize the Model to Python
+        log_file.write('Re-serializing Model JSON.\n')
+        with open(model_json) as json_file:
+            data = json.load(json_file)
+        model = Model.from_dict(data)
+        log_file.write('Model re-serialization successful.\n')
+
+        # translate the model to a radiance folder
+        log_file.write('Translating Model to Radiance folder.\n')
+        rad_fold = model.to.rad_folder(model, folder, folder_type, minimal)
+        log_file.write('Model translation successful.\n')
+        log_file.write('Radiance folder output to: {}'.format(rad_fold))
+    except Exception as e:
+        _logger.exception('Model translation failed.\n{}'.format(e))
+        sys.exit(1)
+    else:
+        sys.exit(0)
+
+
+@translate.command('model-to-rad')
+@click.argument('model-json')
+@click.option('--blk', help='Boolean to note whether the "blacked out" version '
+              'of the geometry should be output, which is useful for direct studies '
+              'and isolation studies of individual apertures.',
+              default=False, show_default=True)
+@click.option('--minimal', help='Boolean to note whether the radiance strings should '
+              'be written in a minimal format (with spaces instead of line breaks).',
+              default=False, show_default=True)
+@click.option('--log-file', help='Optional RAD file to output the RAD string of the '
+              'translation. By default this will be printed out to stdout',
+              type=click.File('w'), default='-')
+def model_to_rad(model_json, blk, minimal, log_file):
+    """Translate a Model JSON file to a Radiance string.
+    \n
+    The resulting strings will include all geometry (Rooms, Faces, Shades, Apertures,
+    Doors) and all modifiers. However, it does not include any states for dynamic
+    geometry and will only write the default state for each dynamic object. To
+    correctly account for dynamic objects, model-to-rad-folder should be used.
+    \n
+    Args:
+        model_json: Full path to a Model JSON file.
+    """
+    try:
+        # check that the model JSON is there
+        assert os.path.isfile(model_json), \
+            'No Model JSON file found at {}.'.format(model_json)
+
+        # re-serialize the Model to Python
+        with open(model_json) as json_file:
+            data = json.load(json_file)
+        model = Model.from_dict(data)
+
+        # translate the model to a rad string and put it in a dictionary
+        model_str, modifier_str = model.to.rad(model, blk, minimal)
+        rad_dict = {'model_str': model_str, 'modifier_str': modifier_str}
+
+        # write out the rad string
+        log_file.write(json.dumps(rad_dict))
```

Comment:
How do you imagine this JSON file to be used? Why not a standard rad file?

Reply:
I think the strength of this method is that it gives you the radiance string in stdout so that the end user can decide what to do with it. You're right that it's weird to have radinace strings in a JSON, though, I'll just join the material sting and the geometry string into one to keep things consistent.